### DISCUSSION TITLE
Do you know what native_longitude and native_latitude mean?

### DISCUSSION THREAD

**Discussion by springfall2008** [timdorr/tesla-api#673] [Q&A]
Are they the home location or something else?

**Comment by GaPhi** [comment]
Probably the raw GPS position before hybridation with
accelerometers/gyroscope thanks to a Kalman filter...
To check the idea, try to drive where there is no GPS signal like a
tunnel...

Le mar. 27 déc. 2022 à 17:55, Trefor Southwell ***@***.***> a
écrit :

**Comment by springfall2008** [comment]
Thanks.

I'm trying to check if the car is at home or away, is there an easy way?

**Comment by GaPhi** [comment]
I use teslamate latitude and longitude data thru MQTT and nodered for any
non-sensitive decisions.
I also combine the WiFi connection of the car on my LAN with its MAC
address if I want the At-home information more surely.


Le mar. 27 déc. 2022 à 19:20, Trefor Southwell ***@***.***> a
écrit :

**Reply by springfall2008** [reply]
Thanks, I'll just compare it against home latitude and longitude within 0.01 for now.

I'm writing some code to charge my car from solar when we have excess energy only

**Comment by GaPhi** [comment]
The best way is to calculate distance between the two points and check the
result is within x meters (x depends on the size of your house/castle
garden 🤡)
Because a distance of 0.01 in latitude or longitude is not the same (cosine
latitude factor on the longitude value)...

Le mar. 27 déc. 2022 à 20:02, Trefor Southwell ***@***.***> a
écrit :

**Comment by alandtse** [comment]
It may be related to [China's GPS system or any nonUS GPS](https://github.com/alandtse/tesla/issues/282#issuecomment-1287837667).
